components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/APIDefinitionFromSwagger12.java

```diff
@@ -0,0 +1,418 @@
+
+
+        APIIdentifier identifier = api.getId();
+
+        APIManagerConfiguration config = ServiceReferenceHolder.getInstance().
+                getAPIManagerConfigurationService().getAPIManagerConfiguration();
+
+        Environment environment = (Environment) config.getApiGatewayEnvironments().values().toArray()[0];
+        String endpoints = environment.getApiGatewayEndpoint();
+        String[] endpointsSet = endpoints.split(",");
+        String apiContext = api.getContext();
+        String version = identifier.getVersion();
+        Set<URITemplate> uriTemplates = api.getUriTemplates();
+        String description = api.getDescription();
+
+        if (endpointsSet.length < 1) {
+            throw new APIManagementException("Error in creating JSON representation of the API" + identifier.getApiName());
+        }
+        if (description == null || description.equals("")) {
+            description = "";
+        } else {
+            description = description.trim();
+        }
+
+
+        Map<String, List<String>> resourceNamepaths = new HashMap<String, List<String>>();
+
+        Map<String, JSONObject> resourceNameJSONs = new HashMap<String, JSONObject>();
+
+        Map<String, List<JSONObject>> resourcePathJSONs = new HashMap<String, List<JSONObject>>();
+
+        List<APIResource> apis = new ArrayList<APIResource>();
+
+        JSONObject mainAPIJson = null;
+
+        try {
+            mainAPIJson = (JSONObject) parser.parse(apiJsonTemplate);
+
+            for (URITemplate template : uriTemplates) {
+                List<Operation> ops;
+                List<Parameter> parameters = null;
+
+                String path = template.getUriTemplate();
+
+                if (path != null && (path.equals("/*") || (path.equals("/")))) {
+                    path = "/*";
+                }
+                List<String> resourcePaths;
+                int resourceNameEndIndex = path.indexOf("/", 1);
+                String resourceName = "/default";
+                if(resourceNameEndIndex != -1) {
+                    resourceName = path.substring(1, resourceNameEndIndex);
+                }
+
+                if(!resourceName.startsWith("/")) {
+                    resourceName = "/" + resourceName;
+                }
+
+                if(resourceNamepaths.get(resourceName) != null) {
+                    resourcePaths = resourceNamepaths.get(resourceName);
+                    if (!resourcePaths.contains(path)) {
+                        resourcePaths.add(path);
+                    }
+                    String httpVerbsStrng = template.getMethodsAsString();
+                    String[] httpVerbs = httpVerbsStrng.split(" ");
+                    for (String httpVerb : httpVerbs) {
+                        final JSONObject operationJson = (JSONObject) parser.parse(operationJsonTemplate);
+                        operationJson.put("method", httpVerb);
```

Comment:
APIDefinitionFromSwagger12 is used with swagger v1.2(In APIM 1.8.0) and removed since it is no longer in use(In APIM 1.9.0)
Instead we have APIDefinitionFromSwagger20 Implemented from same abstract class which implements APIDefinitionFromSwagger12